pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/inv/text/LuceneFSTIndexCreator.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.creator.impl.inv.text;
+
+import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
+import org.apache.lucene.store.OutputStreamDataOutput;
+import org.apache.lucene.util.fst.FST;
+import org.apache.pinot.core.segment.creator.DictionaryBasedInvertedIndexCreator;
+import org.apache.pinot.core.segment.creator.impl.SegmentColumnarIndexCreator;
+import org.apache.pinot.core.util.fst.FSTBuilder;
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedOutputStream;
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+
+public class LuceneFSTIndexCreator implements DictionaryBasedInvertedIndexCreator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SegmentColumnarIndexCreator.class);
+    private final File _fstIndexFile;
+    private final File _dictFile;
+    public static final String FST_INDEX_FILE_EXTENSION = ".lucene.fst";
+    public static final String FST_DICTID_FILE_EXTENSION = ".dict.lucene.fst";
+
+
+    // TODO: Replace this with a memory efficient dict.
+    private final FSTBuilder _fstBuilder;
+    private final Object2IntOpenHashMap<String> _stringValueToIndexMap;
```

Comment:
Why maintaining a separate map inside the creator instead of passing in a dictionary? This map will potentially consume lots of heap memory, and cause memory issue for segment generation.